### PR TITLE
482 set consolidated

### PR DESCRIPTION
**Pull request by matthopson**

### CODE REVIEW

File: shared/src/business/entities/cases/Case.js [hunk 1]
```diff
@@ -1269,4 +1269,15 @@ Case.prototype.canConsolidate = function() {
   return !ineligibleStatusTypes.includes(this.status);
 };
 
+/**
+ * sets lead case id on the current case
+ *
+ * @returns {boolean} true if eligible for consolidation, false otherwise
```
Comment (rachaelparris) [inline shared/src/business/entities/cases/Case.js]: Update jsdoc

File: shared/src/business/useCases/setConsolidatedCaseInteractor.test.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        role: User.ROLES.docketClerk,
+      }),
+      getPersistenceGateway: () => ({
+        getCaseByCaseId: getCaseByCaseIdMock,
+        updateCase: updateCaseMock,
+      }),
+    };
+  });
+
+  it('Should return an Unauthrized error if the user does not have the CONSOLIDATE_CASES permission', async () => {
```
Comment (rachaelparris) [inline shared/src/business/useCases/setConsolidatedCaseInteractor.test.js]: Typo `Unauthrized`

File: web-api/serverless-cases.yml [hunk 3]
```diff
@@ -390,3 +390,15 @@ functions:
             type: COGNITO_USER_POOLS
             authorizerId:
               Ref: ApiGatewayAuthorizer
+
+  updateCaseContext:
```
Comment (rachaelparris) [inline web-api/serverless-cases.yml]: Update this name?

File: shared/src/proxies/setConsolidatedCaseProxy.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+ * @returns {Promise<*>} the promise of the api call
+ */
+exports.setConsolidatedCaseInteractor = ({
+  applicationContext,
+  caseId,
+  leadCaseId,
+}) => {
+  return put({
+    applicationContext,
+    body: leadCaseId,
```
Comment (rachaelparris) [inline shared/src/proxies/setConsolidatedCaseProxy.js]: I _think_ you may need to send this as `body: { leadCaseId }` as your lambda is written. Might be worth checking at least. 😄

File: shared/src/business/entities/cases/Case.test.js [hunk 5]
```diff
@@ -2155,5 +2155,23 @@ describe('Case entity', () => {
...
+        const caseEntity = new Case(
+          {
+            ...MOCK_CASE,
+            preferredTrialCity: 'Birmingham, AL',
+            procedureType: 'regular',
+            status: 'Submitted',
+          },
+          { applicationContext },
+        );
+        const result = await caseEntity.setLeadCase(leadCaseId);
```
Comment (kkoskelin) [inline shared/src/business/entities/cases/Case.test.js]: Probably no `async/await` necessary (though no harm done if we think this might change)

**Review by kkoskelin — APPROVED** [review]

**Review by wbyoko — APPROVED** [review]

**Review by kkoskelin — APPROVED** [review]

File: shared/src/persistence/dynamo/cases/updateCase.js [hunk 6]
```diff
@@ -55,11 +55,16 @@ exports.updateCase = async ({ applicationContext, caseToUpdate }) => {
     }
   }
 
+  let setLeadCase = caseToUpdate.leadCaseId
+    ? { gsi1pk: caseToUpdate.leadCaseId }
```
Comment (matthopson) [inline shared/src/persistence/dynamo/cases/updateCase.js:59]: @codyseibert Would you prefer to see this as `lead-case|${caseToUpdate.leadCaseId}` ? Or is the id sufficient? I assume it makes no difference outside of readability.

Comment (sutt0n) [reply shared/src/persistence/dynamo/cases/updateCase.js:59]: For the sake of readability, I vote for prepending `lead-case|`.

**Review by wbyoko — APPROVED** [review]

**Review by sutt0n — APPROVED** [review]
👍 

The only thing that bothers me is the naming inconsistency between `setConsolidatedCase` versus `addConsolidatedCase` filenames/variables.

**Review by wbyoko — COMMENTED** [review]
does this handle the situation of a case being consolidated with/though a non-lead-already-consolidated case?

**Comment by matthopson** [comment]
@wbyoko Yes, https://github.com/flexion/ef-cms/pull/3785/files#diff-a32d2df72d1a141ceeb3dd898c2fe675R47

or maybe that's not what you meant?

It expects the leadCaseId and any other caseId that it should be consolidated with. If no consolidation attempts have happened previously both the lead and case to consolidate get the leadCaseId param which essentially marks them as consolidated.

File: shared/src/business/useCases/setConsolidatedCaseInteractor.js [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+ * @param {object} providers the providers object
+ * @param {object} providers.applicationContext the application context
+ * @param {object} providers.caseId the id of the case to consolidate
+ * @param {object} providers.leadCaseId the id of the lead case for consolidation
+ * @returns {object} the updated case data
+ */
+exports.setConsolidatedCaseInteractor = async ({
+  applicationContext,
+  caseId,
+  leadCaseId,
```
Comment (wbyoko) [inline shared/src/business/useCases/setConsolidatedCaseInteractor.js]: this should be seen as caseToConsolidateId not leadCaseId

Comment (wbyoko) [inline shared/src/business/useCases/setConsolidatedCaseInteractor.js]: this should be seen as caseToConsolidateId not leadCaseId

**Review by sutt0n — APPROVED** [review]

**Pull request merged**